### ISSUE TITLE
TypeError: Invalid value used as weak map key

### ISSUE THREAD

**Issue by abhiaiyer91** [threepointone/glamor#170]
Hey there Sunil,

After uprading to 2.20.13, I'm getting this error 

`TypeError: Invalid value used as weak map key` on load of my site.

Any clue?

**Comment by threepointone** [comment]
I must have broken something, will check. Do you have some sample css its
breaking on?

**Comment by abhiaiyer91** [comment]
Damn I wish I had some sample css for you, its hard to parse given my Meteorjs errors. Let me look and ill get back to you.

**Comment by threepointone** [comment]
I'm investigating too, will let you know.

**Comment by threepointone** [comment]
My guess is you're sending an actual string as input into css() somewhere.

**Comment by abhiaiyer91** [comment]
Let me verify this brb!

**Comment by abhiaiyer91** [comment]
None of my css calls are doing straight up strings. But my webpack devtools arent being to helpful at `eval` any ideas to help you out?

**Comment by threepointone** [comment]
I'll put in some code to work around this bug, but do please try to figure
out what CSS caused this.

**Comment by abhiaiyer91** [comment]
on it

**Comment by threepointone** [comment]
I published 2.20.16 with a try/catch for the above error, and it logs the arguments to console at least  once. Give it a spin and lemme know, if it fixes your issue, and what it spits out to console for this error.

**Comment by abhiaiyer91** [comment]
Yo sunil, you were right, it was a css fuck up in another module we have

```
failed setting the WeakMap cache for args: data-css-wn814s: ""
this should NOT happen, please file a bug on the github repo.
```

**Comment by abhiaiyer91** [comment]
Thanks so much for putting this log in here. I was wondering if we could do this for any error generated by Glamor? often times its my own doing, but it would be nice to know what blew up since my app is humungous

**Comment by abhiaiyer91** [comment]
Damn, we have a bunch of these in our app. I wonder if the aphrodite shim people have these

` foo ? css(bar) : ''`

cause i believe `null` in aphrodite's `css` blows up

**Comment by threepointone** [comment]
? null, undefined should work fine in `css`, was fixed in #42 and #44. just tried it out too.  

that error message that got logged looks odd... 

I'd be happy to help you debug for a few minutes if you've been able to reproduce this, and can use ngrok or something. Lemme know.

**Comment by threepointone** [comment]
ok, I think I know what's going on, it's quite subtle. Gimme a day or so and I'll fix this.

**Comment by threepointone** [comment]
your current workaround should be to avoid sending those strings.

**Comment by abhiaiyer91** [comment]
Yeah what Im saying is null/undefined did not work in `aphrodite` unless im mistaken. So we had tons of code just returned empty strings. Thus causing this error when we did a sweeping change to use glamor/aphrodite.

Resolving the ternaries to null work out fine for us, just tons of those to go through. Let me know if you need anything else

**Comment by threepointone** [comment]
I got in a fix for you, live in 2.20.19. lemme know if it works.

**Comment by abhiaiyer91** [comment]
Thanks a lot Sunil. will check this out today! Appreciate the speed!

**Comment by threepointone** [comment]
You'll appreciate the speed of the new release more :P The aphrodite shim goes very well with the new WeakMap caches, so your app should get faster for free. Lemme know!

**Comment by threepointone** [comment]
closing this, lemme know if the issue persists. cheers.
